.github/workflows/reusable-python-publish_package.yml

```diff
@@ -0,0 +1,176 @@
+        env:
+            cache_number: 1
+        steps:
+            -   uses: actions/checkout@v4
+                with:
+                    lfs: ${{ inputs.lfs }}
+                    fetch-depth: 0
+            -   name: Set up Python version
+                uses: actions/setup-python@v5
+                with:
+                    python-version: ${{ inputs.python_version }}
+            -   uses: MiraGeoscience/CI-tools/.github/actions/reusable-python-setup_poetry@main
+                name: Setup poetry env
+                if: ${{ inputs.package_manager == 'poetry' }}
+                with:
+                    python_ver: ${{ inputs.python_version }}
+                    cache_number: ${{ env.cache_number}}
+                    runner_os: ${{ runner.os }}
+                    extra_repository: ${{ inputs.extra_repository }}
+                    repo_user: ${{ secrets.EXTRA_PYPI_REPO_USER }}
+                    repo_token: ${{ secrets.EXTRA_PYPI_REPO_TOKEN }}
+            -   uses: MiraGeoscience/CI-tools/.github/actions/reusable-python-setup_hatch@main
+                name: Setup hatch env
+                if: ${{ inputs.package_manager == 'hatch' }}
+                with:
+                    cache_number: ${{ env.cache_number}}
+                    runner_os: ${{ runner.os }}
+            -   name: Build package
+                run: |
+                    if ${{ inputs.package_manager == 'poetry' }}; then
+                        poetry build
+                    elif ${{ inputs.package_manager == 'hatch'}}; then
+                        hatch build
+                    else
+                        echo "Unknown pypi package manager. ${{ inputs.package_manager }} is not part of ['hatch', 'poetry']."
+                        exit 1
+                    fi
+            -   name: Publish
+                run: |
+                    MIRA_REPO_CONFIG=$(echo '${{ secrets.MIRA_REPO_SECRETS }}' | base64 -d)
+                    repository_list=$(echo '${{ inputs.repo_names }}' | jq -r '.[]')
+                    for repo_name in ${repository_list}; do
+                        repo_user=$(echo ${MIRA_REPO_CONFIG} | jq -r --arg repo_name "$repo_name" '.repositories[] | select(.name == $repo_name) | .username')
+                        repo_token=$(echo ${MIRA_REPO_CONFIG} | jq -r --arg repo_name "$repo_name" '.repositories[] | select(.name == $repo_name) | .token')
+                        repo_url=$(echo ${MIRA_REPO_CONFIG} | jq -r --arg repo_name "$repo_name" '.repositories[] | select(.name == $repo_name) | .url')
+                        if ${{ inputs.package_manager == 'poetry' }}; then
+                            poetry config repositories.${repo_name} ${repo_url} && \
+                            poetry publish -r ${repo_name} -u ${repo_user} -p ${repo_token}
+                        elif ${{ inputs.package_manager == 'hatch'}}; then
+                            hatch publish -r ${repo_url} -u ${repo_user} -a ${repo_token}
```

Comment:
**thought (informative, non-blokcing)**: alternatively, could set env vars for hatch:

- `HATCH_INDEX_REPO` 
- `HATCH_INDEX_USER`
- `HATCH_INDEX_AUTH`

and simply call: `hatch publish`

Reply:
We could do it, I agree. But, what is the purpose given we have to redefine theses env variables each time we want to publish a package in a new repository ? (analyst-pypi-dev, analyst-pypi, pypi, pypi-unofficial, etc) ?

Reply:
I have just mentioned it for discussion in case there was value. I agree though it is better for maintainability of this script if the different ways of publishing all follow the same pattern.

Not that it matters in the scope of a github action, but note that command line reveals the value of the variables passed as parameters:

`python -c "import sys;print(sys.argv[1]);sleep(5000)" %val%`
Then, in task manager:
![image](https://github.com/user-attachments/assets/70875e88-8012-49ab-aa17-58ba019773ff)

Reply:
BTW, variables for Poetry would be:

- `POETRY_PYPI_TOKEN_${repo_name}=my-token`
- `POETRY_HTTP_BASIC_${repo_name}_USERNAME=<username>`
- `POETRY_HTTP_BASIC_${repo_name}_PASSWORD=<password>`

Reply:
anyway, good as is for now. Maybe just apply changes for paranoïd mode 😉  using `$(echo ${val})`